packages/agent/src/invitation/handler.ts

```diff
@@ -199,26 +201,15 @@ export const receiveCredentialFromOpenId4VciOffer = async ({
         didMethod = 'key'
       }
 
-      let key: Key | undefined = undefined
-
-      // For P-256 we first try secure enclave
-      if (keyType === KeyType.P256) {
-        key = await agent.wallet
-          .createKey({
-            keyType,
-            keyBackend: KeyBackend.SecureElement,
-          })
-          .catch((e) => {
-            agent.config.logger.warn('Could not create a key in the secure element', e as Record<string, unknown>)
-            return agent.wallet.createKey({
-              keyType,
-            })
-          })
-      } else {
-        key = await agent.wallet.createKey({
-          keyType,
-        })
-      }
+      // TODO: require support for mDoc namespace here as well
+      const vct = resolvedCredentialOffer.offeredCredentialConfigurations[offeredCredentialToRequest.id].vct
```

Comment:
Use `supportedCredentialId` that is passed to this callback. This will help us when we add mdoc as well and request multiple creds.

Also -- vct is not always defined. So this check should be something like:

```
let shouldKeyBeHardwareBacked = false

const offeredCredentialConfiguration = offeredCredentialConfigurations[supportedCredentialId]

// TODO: use the enum for format, not sure what its' called)
if (offeredCredentialConfiguration.format === 'vc+sd-jwt' && pidSchemes?.sdJwtVcVcts.includes(offeredCredentialConfiguration.vct)) {
  shouldKeyBeHardwareBacked = true
}

```

Finally, I'm not sure why we need to call it pidSchemes. That's more implementation specific. Why not call it useHardwareKeySchemes? Or just pass `useHardwareKey: true/false` to this method? In receive pid use case we pass true, by default it's false